zim/plugins/base/imagegenerator.py

```diff
@@ -140,13 +141,13 @@ def __init__(self, notebook, page, generator, attrib, data):
 			self.image_file.remove()
 
 	def _new_image_file(self):
-		cache_dir = self.notebook.folder.folder('_images')
+		cache_dir = self.notebook.cache_dir.folder('images')
 		if self.data:
 			content = []
 			for k, v in sorted(self.attrib.items()):
-				content.extend([k, v])
+				content.extend([k, str(v)])
 			content.append(self.data)
-			basename = hashlib.md5(''.join(content).encode()).hexdigest() + self.generator.imagefile_extension
+			basename = hashlib.sha224(''.join(content).encode()).hexdigest() + self.generator.imagefile_extension
```

Comment:
I'd leave md5 here; as you said it's not a security concern and the hash is only used to check if the image exists in cache -- no need to spend the watts